### PR TITLE
Feat: Add ActivityJoinEvent

### LINKED ISSUE #89: onActivityJoin isn't fired
as title, it seem that the event `onActivityJoin` was never fired?

I have tested it on ae841453bd (master latest) and 70b1e9dda6189cd9fbdce113527a915f892bc154 (java-impl latest) and both seems never fire the event?

### PR DESCRIPTION
**Pull request by RedTeaDev**
Fixes #89

### CODE REVIEW

File: src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        public void handle(Command command, DiscordUser user)
+        {
+            core.log(LogLevel.VERBOSE, "Secret: " + command.getData().getAsJsonObject().get("secret").getAsString());
+            core.getEventAdapter().onActivityJoin(command.getData().getAsJsonObject().get("secret").getAsString());
+        }
+
+        @Override
+        public Class<?> getDataClass()
+        {
+            return DiscordUser.class;
```
Comment (JnCrMx) [inline src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java:27]: Is there a reason that DiscordUser is the data class?
It seems like it is not used in `handle(...)`.

File: src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        public Handler(Core.CorePrivate core)
+        {
+            super(core);
+        }
+
+        @Override
+        public void handle(Command command, DiscordUser user)
+        {
+            core.log(LogLevel.VERBOSE, "Secret: " + command.getData().getAsJsonObject().get("secret").getAsString());
```
Comment (JnCrMx) [inline src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java:20]: Do we need to log this?

File: src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            core.getEventAdapter().onActivityJoin(command.getData().getAsJsonObject().get("secret").getAsString());
```
Comment (JnCrMx) [inline src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java:21]: I would personally not get the secret directly from the JSON object, but instead use the existing data mechanism.

e.g.:
```java
public static class Data {
    String secret;
}

public static class Handler extends EventHandler<Data> {

    @Override
    public void handle(Command command, Data data) {
        // use data.token
    }

    @Override
    public Class<?> getDataClass() {
        return Data.class;
    }
}

**Review by JnCrMx — CHANGES_REQUESTED** [review]

File: src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java (hunk 2 above)
Comment (RedTeaDev) [reply src/main/java/de/jcm/discordgamesdk/impl/events/ActivityJoinEvent.java:20]: oh i forgot to remove that when trying to add the feature, sorry, will push a commit removing that later

**Review by JnCrMx — APPROVED** [review]
lgtm now, a few unused imports due to the changes, but I can clean them up later

**Pull request merged**